### ISSUE TITLE
unsigned triggers PRS

### ISSUE THREAD

**Issue by covracer** [sqlfluff/sqlfluff#6844]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```python
In [66]: from sqlfluff import fix; from sqlfluff.core.config import FluffConfig

In [70]: result = fix(
    ...:     'CREATE TABLE "wellplated_format" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "bottom_row" varchar(1) NOT NULL, "right_column" smallint unsigned NOT NULL CHECK ("right_colum
    ...: n" >= 0));',
    ...:     'sqlite',
    ...:     config=FluffConfig({'core': {'dialect': 'sqlite', 'max_line_length': 100}}),
    ...: )

In [71]: result
Out[71]: 'CREATE TABLE "wellplated_format" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "bottom_row" varchar(1) NOT NULL, "right_column" smallint unsigned NOT NULL CHECK ("right_column" >= 0));'

In [74]: for line in result.splitlines():
    ...:     print(len(line))
    ...:
188
```

### Expected Behaviour

Newlines should be added in order to get each line below `max_line_length`

### Observed Behaviour

Input is unchanged. If I run lint instead of fix, I see `PRS`: "Line 1, Position 1: Found unparsable section".

### How to reproduce
[edited to minimize reproducer and highlight workaround]
```python
from sqlfluff import lint
smallest = 'CREATE TABLE "wellplated_format" ("right_column" smallint unsigned NOT NULL CHECK ("right_column" >= 0));\n'
print(smallest, lint(smallest, 'sqlite'))  # PRS is bad
workaround = smallest.replace('unsigned', '/*unsigned*/')
print(workaround, lint(workaround, 'sqlite'))  # LT05 is good
```

### Dialect

sqlite

### Version

3.4.0

### Configuration

```python
{'core': {'dialect': 'sqlite', 'max_line_length': 100}
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by covracer** [comment]
After some experimentation, it looks like `unsigned` is the problem. If I take it out, `fix()` works as expected and `lint()` stops returning `Found unparsable section`
